src/persistence/Elsa.Persistence.EntityFrameworkCore/DbContexts/ElsaContext.cs

```diff
@@ -46,6 +52,29 @@ private void ConfigureWorkflowDefinitionVersion(ModelBuilder modelBuilder)
             entity.HasMany(x => x.Connections).WithOne(x => x.WorkflowDefinitionVersion);
         }
 
+        private void ConfigureCustomSchema(ModelBuilder modelBuilder)
+        {
+            if (Services != null && _dbContextCustomSchema == null)
+            {
+                using (var scope = Services.BuildServiceProvider().CreateScope())
+                {
+                    _dbContextCustomSchema = scope.ServiceProvider.GetService<IDbContextCustomSchema>();
+                }
+            }
+            if (_dbContextCustomSchema != null && _dbContextCustomSchema.UseCustomSchema)
+            {
+                modelBuilder.HasDefaultSchema(_dbContextCustomSchema.CustomDefaultSchema);
+
+                // Apply the custom mapping to suport the non-default schema to the types in used in this context.
+                modelBuilder.ApplyConfiguration(new ActivityDefinitionEntityConfiguration(_dbContextCustomSchema));
```

Comment:
All of these `*EntityConfiguration` classes have exactly the same implementation, where only the table name varies. Therefore it seems more DRY if you create just one class called e.g. `CustomSchemaEntityConfiguration`, taking the table name as a second ctor argument.